File: Views/src/components/financial/components/bank-account-cards/dto/card-dto.ts
import { RootBase } from "src/shared/entities-dtos/root-base";
import { BankAccountDto } from "./bank-account-dto";
import { CreditCardLimitOperationDto } from "./credit-card-limit-operation-dto";
import { TypeCardDtoEnum } from "./enums/type-card-dto.enum";
import { CreditCardExpenseInvoiceDto } from "../../credit-card-fixed-expenses/components/list -invoices/dto/credit-card-expense-invoice-dto";

export class CardDto extends RootBase{
  holder: string;
  flag:string;
  creditLimit: number;
  creditCardLimitOperation:CreditCardLimitOperationDto;
  type: TypeCardDtoEnum;
  number: string;
  cvc: number;
  description:string;
  validate: Date;
  closingDate:Date;
  expiresDate:Date;
  bankAccountId:number;
  bankAccount:BankAccountDto;
  creditCardExpensesInvoices:CreditCardExpenseInvoiceDto[]
}

File: Application/Services/Operations/Finances/Dtos/Bank/CardDto.cs
using System;
using System.Collections.Generic;
using Application.Services.Operations.Authentication.Dtos;
using Application.Services.Operations.Finances.Dtos.CreditCardExpenses;
using Application.Services.Operations.Finances.Dtos.Enums;
using Application.Services.Operations.Main.Companies.Dtos;
using Application.Services.Shared.Dtos;

namespace Application.Services.Operations.Finances.Dtos.Bank
{
    public class CardDto : RootBaseDto
    {
        public string Holder { get; set; }
        public string Flag { get; set; }
        public decimal CreditLimit { get; set; }
        public TypeCardEnumDto Type { get; set; }
        public string Number { get; set; }
        public int CVC { get; set; }
        public string Description { get; set; }
        public DateTime Validate { get; set; }
        public DateTime ClosingDate { get; set; }
        public DateTime ExpiresDate { get; set; }
        public int BankAccountId { get; set; }
        public BankAccountDto BankAccount { get; set; }
        public CreditCardLimitOperationDto CreditCardLimitOperation { get; set; }
        public List<CreditCardExpenseInvoiceDto> CreditCardExpensesInvoices { get; set; }

    }
}